core/src/main/java/com/google/errorprone/bugpatterns/UnnecessaryParentheses.java

```diff
@@ -44,7 +46,10 @@
   @Override
   public Description matchParenthesized(ParenthesizedTree tree, VisitorState state) {
     ExpressionTree expression = tree.getExpression();
-    if (state.getPath().getParentPath().getLeaf() instanceof StatementTree) {
+    Tree parent = state.getPath().getParentPath().getLeaf();
+    if (parent instanceof StatementTree) {
+      return NO_MATCH;
+    }else if (parent instanceof SwitchExpressionTree){
```

Comment:
I'm going to change this a bit when I merge the PR, because we want to avoid referencing `SwitchExpressionTree` because that class is only available in JDK >= 12, and we want to remain compatible with JDK 8 for now.

Reply:
Sounds Good. As long as you don't need anything else from me I'll let you take it from here.